src/c-writer.cc

```diff
@@ -584,12 +648,25 @@ void CWriter::Write(const ExternalPtr& name) {
   }
 }
 
+void CWriter::Write(const ExternalInstancePtr& name) {
+  bool is_import = import_syms_.count(name.name) != 0;
+  if (is_import) {
+    Write("module_instance->", GetGlobalName(name.name));
+  } else {
+    Write("&module_instance->", GetGlobalName(name.name));
```

Comment:
This is in the implementation of Write(ExternalInstancePtr), so the goal is to write a pointer. The imported elements are already pointers, so for those it can just write the name of the element. For something declared in the module itself, it needs to take the address of the element with the `&` operator.